pkg/R/utils.R

```diff
@@ -164,11 +176,14 @@ reserialize <- function(rdd) {
   if (!inherits(rdd, "RDD")) {
     stop("Argument 'rdd' is not an RDD type.")
   }
-  if (rdd@env$serialized) {
-    return(rdd)
-  } else {
+  if (rdd@env$serializedMode != "byte") {
+    if (rdd@env$serializedMode == "row") {
+      rdd@env$serialzedMode == "byte"
```

Comment:
I don't think I'm following, actually. Since the RDD constructor uses `byte` as the default argument for `serializedMode`, wouldn't it necessarily be the case that any RDD you interact with in SparkR would be guaranteed to have a value for `serializedMode`?  I guess I don't understand how we could have an RDD S4 class object without necessarily having a JRDD backing it.